modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/FSMCallerImpl.java

```diff
@@ -542,6 +546,8 @@ private void doCommitted(final long committedIndex) {
             if (iterImpl.hasError()) {
                 setError(iterImpl.getError());
                 iterImpl.runTheRestClosureWithError();
+            } else if (shuttingDown) {
+                iterImpl.runTheRestClosureWithShutdownException();
             }
             final long lastIndex = iterImpl.getIndex() - 1;
```

Comment:
Hmm, seems that there are some minor leaks. Let's say we have concurrent disasterRecovery procedure that is trying to find out most up-to-date raft node (node with max log index). It reads "incorrect" lastLogId and returns it to the requester. It look like a minor suboptimality of disaster recovery procedure. Though it might be solved by returning last persisted log id if node is shutting down.

Reply:
`lastAppliedId` will correspond to last command that was actually applied, so it seems that it will be correct.

Also, does disaster recovery procedure look at applied index? I though it looked at last index in log (regardless of whether it was committed or applied).See `RaftServer#raftNodeIndex()`

Reply:
Yes, you are right it does look at last log index in logManager.